nixos/modules/virtualisation/incus.nix

```diff
@@ -0,0 +1,251 @@
+    virtualisation.incus = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          This option enables incusd, a daemon that manages
+          containers. Users in the "incus-admin" group can interact with
+          the daemon (e.g. to start or stop containers) using the
+          {command}`incus` command line tool, among others.
+        '';
+      };
+
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.incus;
+        defaultText = lib.literalExpression "pkgs.incus";
+        description = lib.mdDoc ''
+          The incus package to use.
+        '';
+      };
+
+      lxcPackage = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.lxc;
+        defaultText = lib.literalExpression "pkgs.lxc";
+        description = lib.mdDoc ''
+          The LXC package to use with Incus (required for AppArmor profiles).
+        '';
+      };
+
+      preseed = lib.mkOption {
+        type = lib.types.nullOr (
+          lib.types.submodule { freeformType = preseedFormat.type; }
+        );
+
+        default = null;
+
+        description = lib.mdDoc ''
+          Configuration for Incus preseed, see
+          <https://linuxcontainers.org/incus/docs/main/howto/initialize/#non-interactive-configuration>
+          for supported values.
+
+          Changes to this will be re-applied to Incus which will overwrite existing entities or create missing ones,
+          but entities will *not* be removed by preseed.
+        '';
+
+        example = lib.literalExpression ''
+          {
+            networks = [
+              {
+                name = "incusbr0";
+                type = "bridge";
+                config = {
+                  "ipv4.address" = "10.0.100.1/24";
+                  "ipv4.nat" = "true";
+                };
+              }
+            ];
+            profiles = [
+              {
+                name = "default";
+                devices = {
+                  eth0 = {
+                    name = "eth0";
+                    network = "incusbr0";
+                    type = "nic";
+                  };
+                  root = {
+                    path = "/";
+                    pool = "default";
+                    size = "35GiB";
+                    type = "disk";
+                  };
+                };
+              }
+            ];
+            storage_pools = [
+              {
+                name = "default";
+                driver = "dir";
+                config = {
+                  source = "/var/lib/incus/storage-pools/default";
+                };
+              }
+            ];
+          }
+        '';
+      };
+
+      socketActivation = lib.mkEnableOption (
+        lib.mdDoc ''
+          Use socket-activation to start incus.service. Enabling this option
+          will stop the service from starting automatically on boot.
+        ''
+      );
```

Comment:
Have a look at the implementation of `mkEnableOption`: https://github.com/NixOS/nixpkgs/blob/c5ada8b4ca8975d380020a085425a2cc7bbbb300/lib/options.nix#L98C5-L98C5

It automatically prepends *Whether to enable* to your string, which would make this become *Whether to enable Use socket-activation to start ....*

Reply:
Thanks, I was unaware of this prepend.